### PR TITLE
python3Packages.sourmash: 4.8.4 -> 4.8.11

### PR DESCRIPTION
**Pull request by luizirber**
## Description of changes

Diff: https://github.com/sourmash-bio/sourmash/compare/v4.8.4...v4.8.11

Changelog: https://github.com/sourmash-bio/sourmash/releases/tag/v4.8.11

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/sourmash/default.nix [hunk 1]
```diff
@@ -20,24 +21,29 @@
...
 
   cargoDeps = rustPlatform.fetchCargoTarball {
     inherit src;
     name = "${pname}-${version}";
-    hash = "sha256-HisWvJgx15OfYoMzzqYm1JyY1/jmGXBSZZmuNaKTDjI=";
+    hash = "sha256-MY6KrXmFAlL3RK7BFNQrte4XiNdLisSkVh035MfTy+A=";
   };
 
+  preConfigure = lib.optionalString stdenv.isDarwin ''
+    export MACOSX_DEPLOYMENT_TARGET=10.14
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sourmash/default.nix:40]: Can we get this version from some darwin package or so?

Comment (luizirber) [reply pkgs/development/python-modules/sourmash/default.nix]: I moved this up to `pkgs/top-level/python-packages.nix`, and used the `overrideSDK` method to declare the min version:
```
  sourmash = callPackage ../development/python-modules/sourmash {
    stdenv = if pkgs.stdenv.isDarwin then pkgs.overrideSDK pkgs.stdenv {
      darwinMinVersion = "10.14";
    } else pkgs.stdenv;
  };
```

Hopefully this is better and less prone to errors in the future

**Review by natsukium — APPROVED** [review]

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -14387,7 +14387,11 @@ self: super: with self; {
 
   soupsieve = callPackage ../development/python-modules/soupsieve { };
 
-  sourmash = callPackage ../development/python-modules/sourmash { };
+  sourmash = callPackage ../development/python-modules/sourmash {
+    stdenv = if pkgs.stdenv.isDarwin then pkgs.overrideSDK pkgs.stdenv {
+      darwinMinVersion = "10.14";
+    } else pkgs.stdenv;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:14393]:
```suggestion
    stdenv = if stdenv.isDarwin then overrideSDK stdenv {
      darwinMinVersion = "10.14";
    } else stdenv;
```

File: pkgs/development/python-modules/sourmash/default.nix [hunk 3]
```diff
@@ -21,24 +22,27 @@
 
 buildPythonPackage rec {
   pname = "sourmash";
-  version = "4.8.4";
+  version = "4.8.10";
   format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sourmash/default.nix:26]:
```suggestion
  pyproject = true;
```

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (luizirber) [reply pkgs/top-level/python-packages.nix]: Moved back into package, followed the example from [renovate](https://github.com/NixOS/nixpkgs/blob/1e4d24952a0671a849ad5d61bcead24e2c398458/pkgs/by-name/re/renovate/package.nix#L17-L21)

File: pkgs/development/python-modules/sourmash/default.nix [hunk 4]
```diff
@@ -19,26 +21,34 @@
   pytestCheckHook,
 }:
 
+let
+
+  stdenv' = if stdenv.isDarwin then overrideSDK stdenv { darwinMinVersion = "10.14"; } else stdenv;
+
+in
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sourmash/default.nix:28]:
```suggestion
```

File: pkgs/development/python-modules/sourmash/default.nix [hunk 5]
```diff
@@ -19,26 +21,34 @@
...
 buildPythonPackage rec {
   pname = "sourmash";
-  version = "4.8.4";
-  format = "pyproject";
-  disabled = pythonOlder "3.8";
+  version = "4.8.11";
+  pyproject = true;
+  disabled = pythonOlder "3.9";
+
+  stdenv = stdenv';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sourmash/default.nix:35]:
```suggestion
  stdenv = if stdenv.isDarwin then overrideSDK stdenv { darwinMinVersion = "10.14"; } else stdenv;
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**